dev/com.ibm.ws.jaxrs.2.x.monitor/src/com/ibm/ws/jaxrs/monitor/JaxRsMonitorFilter.java

```diff
@@ -169,15 +169,20 @@ public void filter(ContainerRequestContext reqCtx, ContainerResponseContext resp
 
 				// Figure out min/max
 				if (elapsedTime >= 0) {
-					synchronized (this) {
-						if (elapsedTime > stats.getMinuteLatestMaximumDuration()) {
-							stats.updateMinuteLatestMaximumDuration(elapsedTime);
+					long minuteLatestMaximumDuration = stats.getMinuteLatestMaximumDuration();
+					while (elapsedTime > minuteLatestMaximumDuration) {
+						if (stats.compareAndUpdateMinuteLatestMaximumDuration(minuteLatestMaximumDuration, elapsedTime)) {
+								break;
 						}
+						minuteLatestMaximumDuration = stats.getMinuteLatestMaximumDuration();
+					}
 
-						if (elapsedTime < stats.getMinuteLatestMinimumDuration()
-						        || stats.getMinuteLatestMinimumDuration() == 0L) {
-							stats.updateMinuteLatestMinimumDuration(elapsedTime);
+					long minuteLatestMinimumDuration = stats.getMinuteLatestMinimumDuration();
+					while (elapsedTime < minuteLatestMinimumDuration || minuteLatestMinimumDuration == 0L) {
```

Comment:
I kind of want to surround this while with:

```
if (!(elapsedTime == 0L && minuteLatestMinimumDuration == 0L)) {
```

Reply:
I added the if check.